src/redux/actions/projectActions.ts

```diff
@@ -88,6 +90,20 @@ export function saveProject(project: IProject)
     };
 }
 
+export function updateProjectTagsFromFiles(project: IProject)
+    : (dispatch: Dispatch, getState: () => IApplicationState) => Promise<void> {
+    return async (dispatch: Dispatch, getState: () => IApplicationState) => {
+        const updatedProject = Object.assign({}, project);
+        updatedProject.tags = [];
+        const projectService = new ProjectService();
+        const storageProvider = StorageProviderFactory.createFromConnection(project.sourceConnection);
+        await projectService.getTagsFromPreExistingFieldFile(updatedProject, storageProvider);
+        await projectService.getTagsFromPreExistingLabelFiles(updatedProject, storageProvider);
+        await projectService.setColorsForUpdatedTags(project, updatedProject);
```

Comment:
add a new public updateProjectFromFiles(project): IProject to projectService
and call these function internally and make them private.